dask/dataframe/io/parquet/core.py

```diff
@@ -553,31 +553,55 @@ def to_parquet(
     # Use i_offset and df.npartitions to define file-name list
     filenames = ["part.%i.parquet" % (i + i_offset) for i in range(df.npartitions)]
 
-    # write parts
-    dwrite = delayed(engine.write_partition)
-    parts = [
-        dwrite(
-            d,
+    # Construct IO graph
+    dsk = {}
+    name = "to-parquet-" + tokenize(
+        df,
+        fs,
+        path,
+        append,
+        ignore_divisions,
+        partition_on,
+        division_info,
+        index_cols,
+        schema,
+    )
+    part_tasks = []
+    kwargs_pass["fmd"] = meta
+    kwargs_pass["compression"] = compression
+    kwargs_pass["index_cols"] = index_cols
+    kwargs_pass["schema"] = schema
+    for d, filename in enumerate(filenames):
+        dsk[(name, d)] = (
+            _write_partition,
+            engine.write_partition,
+            (df._name, d),
             path,
             fs,
             filename,
             partition_on,
             write_metadata_file,
-            fmd=meta,
-            compression=compression,
-            index_cols=index_cols,
-            schema=schema,
-            **kwargs_pass,
+            kwargs_pass,
         )
```

Comment:
Thanks for doing this.  In general using delayed is a bit of a crutch.  Moving core collections to layers with raw dicts is generally always better.  In the future, I imagine that we'll work on moving things to abstract layers :+1: